### DISCUSSION TITLE
Insatalling WASM deps: header files not found?

### DISCUSSION THREAD

**Discussion by madig** [harfbuzz/harfbuzz#4338] [Q&A]
I followed the instructions on https://github.com/harfbuzz/harfbuzz/blob/main/docs/wasm-shaper.md#enabling-the-wasm-shaper-when-building-harfbuzz to compile wasm stuff, but compilation fails because it can't find `wasm_export.h`? Did I miss a step somewhere? Installing the build directory just installs iwasm and libiwasm.so, so I'm not even sure how to show it to meson, except by fiddling with the build files?

**Comment by behdad** [comment]
You mean the cmake install in wasm-micro-runtime doesn't install any headers? It sure does for me. See eg.
https://github.com/bytecodealliance/wasm-micro-runtime/issues/1990

**Reply by madig** [reply]
Ah. The mistake was that I went into `product-mini/platforms/linux`, I need to do the cmake dance from the root level. The docs are misleading, unless I'm misleading myself?

**Comment by behdad** [comment]
Humm, yeah, that cd seems redundant in the docs. Mine works with cmake from toplevel. @simoncozens

**Reply by madig** [reply]
Sent a PR.

Now battling with the utils not finding libiwasm.so when it's right there in /usr/local/lib, but that may be an error on my side...

**Reply by behdad** [reply]
Yeah I don't know why that is. I do:
```
export LD_LIBRARY_PATH=/usr/local/lib
```

**Reply by devosb** [reply]
Does running `sudo ldconfig` (thus updating the linker cache) help utils find the library? Assuming that `/usr/local/lib` is listed somewhere under `/etc/ld.so.conf*`?

**Reply by behdad** [reply]
I think on a 64bit machine only `/usr/local/lib64` is searched by default. This seems to me like a bug in wasm-micro-runtime that installs 64bit binaries in `/usr/local/lib`.
